solara/server/app.py

```diff
@@ -110,45 +138,26 @@ def add_path():
                     "can avoid this ambiguity."
                 )
 
-        elif self.name.endswith(".py"):
-            self.type = AppType.SCRIPT
+        elif self.type == AppType.SCRIPT:
             add_path()
             # manually add the script to the watcher
             reload.reloader.watcher.add_file(self.path)
-            self.directory = self.path.parent.resolve()
             initial_namespace = {
                 "__name__": "__main__",
             }
             with reload.reloader.watch():
                 routes = [solara.autorouting._generate_route_path(self.path, first=True, initial_namespace=initial_namespace)]
-        elif self.name.endswith(".ipynb"):
-            self.type = AppType.NOTEBOOK
+        elif self.type == AppType.NOTEBOOK:
```

Comment:
This case and `self.type == AppType.SCRIPT` are almost identical, maybe their cases here can be merged?

Reply:
Good catch, but I'd rather keep it like this, I don't think the code will look cleaner in the end